dali/util/shared_mem.cc

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <errno.h>
+#include <fcntl.h> /* For O_* constants */
+#include <sys/mman.h>
+#include <sys/stat.h> /* For mode constants */
+#include <unistd.h>
+#include <ctime>
+#include <exception>
+#include <random>
+#include <sstream>
+#include <string>
+
+#include "dali/util/shared_mem.h"
+
+namespace dali {
+namespace python {
+// Abstract away the fact that shm_open requires filename.
+ShmFdWrapper::ShmFdWrapper(int fd) : fd{fd} {}
+
+ShmFdWrapper::ShmFdWrapper() {
+  static std::minstd_rand rand_suffix((unsigned)time(NULL) * getpid());
+  std::string name;
+  int sanity_counter = 0;
+  do {
+    std::stringstream ss;
+    ss << "/nvidia_dali_" << rand_suffix();
```

Comment:
Why not https://man7.org/linux/man-pages/man3/mktemp.3.html ?

Reply:
Hmm, sounds like exactly what I need. On the other hand though in section bugs is says never use me. :D

Reply:
Then maybe https://man7.org/linux/man-pages/man3/mkstemp.3.html ?

Reply:
It seems to work! The only thing that bothers me a little with this approach is that I need to specify full path in mkstemp to /dev/shm which is the place where shm_open would create a file. I wonder if that's portable enough for us.

Reply:
I think it should be fine.